60.md

```diff
@@ -0,0 +1,188 @@
+* `unit` - Base unit of the wallet (e.g. "sats", "usd", etc).
+* `name` - Optional human-readable name for the wallet.
+* `description` - Optional human-readable description of the wallet.
+* `balance` - Optional best-effort balance of the wallet that can serve as a placeholder while an accurate balance is computed from fetching all unspent proofs.
+* `privkey` - Private key used to unlock P2PK ecash. MUST be stored encrypted in the `.content` field. **This is a different private key exclusively used for the wallet, not associated in any way to the user's nostr private key** -- This is only used when receiving funds from others, described in NIP-61.
+
+Any tag, other than the `d` tag, can be [[NIP-44]] encrypted into the `.content` field.
+
+### Deleting a wallet event
+Due to PRE being hard to delete, if a user wants to delete a wallet, they should empty the event and keep just the `d` identifier and add a `deleted` tag.
+
+## Token Event
+Token events are used to record the unspent proofs that come from the mint.
+
+There can be multiple `kind:7375` events for the same mint, and multiple proofs inside each `kind:7375` event.
+
+```jsonc
+{
+    "kind": 7375,
+    "content": nip44_encrypt({
+        "mint": "https://stablenut.umint.cash",
+        "proofs": [
+            {
+                "id": "005c2502034d4f12",
+                "amount": 1,
+                "secret": "z+zyxAVLRqN9lEjxuNPSyRJzEstbl69Jc1vtimvtkPg=",
+                "C": "0241d98a8197ef238a192d47edf191a9de78b657308937b4f7dd0aa53beae72c46"
+            }
+        ]
+    }),
+    "tags": [
+        [ "a", "37375:<pubkey>:my-wallet" ]
+    ]
+}
+```
+
+`.content` is a [[NIP-44]] encrypted payload storing the mint and the unencoded proofs.
+* `a` an optional tag linking the token to a specific wallet.
+
+### Spending proofs
+When one or more proofs of a token are spent, the token event should be [[NIP-09]]-deleted and, if some proofs are unspent from the same token event, a new token event should be created rolling over the unspent proofs and adding any change outputs to the new token event.
+
+## Spending History Event
+Clients SHOULD publish `kind:7376` events to create a transaction history when their balance changes.
+
+```jsonc
+{
+    "kind": 7376,
+    "content": nip44_encrypt([
+        [ "direction", "in" ], // in = received, out = sent
+        [ "amount", "1", "sats" ],
+        [ "e", "<event-id-of-spent-token>", "<relay-hint>", "created" ],
+    ]),
+    "tags": [
+        [ "a", "37375:<pubkey>:my-wallet" ],
+    ]
+}
+```
+
+* `direction` - The direction of the transaction; `in` for received funds, `out` for sent funds.
+* `a` - The wallet the transaction is related to.
+
+Clients MUST add `e` tags to create references of destroyed and created token events along with the marker of the meaning of the tag:
+* `created` - A new token event was created.
+* `destroyed` - A token event was destroyed.
+* `redeemed` - A [[NIP-61]] nutzap was redeemed.
```

Comment:
Would it make sense to also log outgoing zaps? When balance changes because of "out" it could be due to the user e.g. melting tokens to lightning, but it could also be dude to sending a zap. (as in "redeemed" but in the other direction)